configure.ac

```diff
@@ -1436,53 +1609,63 @@ dnl ***************************************************************************
-		IVYKIS_NO_LIBTOOL_LIBS="-Wl,${WHOLE_ARCHIVE_OPT} -L\$(top_builddir)/lib/ivykis/src/.libs -livykis -Wl,${NO_WHOLE_ARCHIVE_OPT}"
+        # LIBS to use when libtool is not applicable (when linking the main syslog-ng executable in mixed linking mode)
+        IVYKIS_NO_LIBTOOL_LIBS="-Wl,${WHOLE_ARCHIVE_OPT} -L\$(top_builddir)/lib/ivykis/src/.libs -livykis -Wl,${NO_WHOLE_ARCHIVE_OPT}"
 
-                # internal ivykis is linked statically, io_uring might be available and linked against too, we have to use it as well
-                AC_SEARCH_LIBS([io_uring_queue_init], [uring])
-		AC_DEFINE(HAVE_IV_WORK_POOL_SUBMIT_CONTINUATION, 1)
-	else
-		AC_MSG_ERROR([Internal ivykis sources not found in lib/ivykis. This is a hard dependency, unable to build syslog-ng without them.])
-	fi
+        # internal ivykis is linked statically, io_uring might be available and linked against too, we have to use it as well
+        AC_SEARCH_LIBS([io_uring_queue_init], [uring], [IO_URING_LIBS="-luring"])
+
+        # internal version surely has iv_work_pool_submit_continuation
+        AC_DEFINE(HAVE_IV_WORK_POOL_SUBMIT_CONTINUATION, 1)
+    else
+        AC_MSG_ERROR([Internal ivykis sources not found in lib/ivykis. This is a hard dependency, unable to build syslog-ng without them.])
+    fi
 else
-	with_ivykis="system"
-	PKG_CHECK_MODULES(IVYKIS, ivykis >= $IVYKIS_MIN_VERSION,,)
-	PKG_CHECK_MODULES(IVYKIS_VERSION_CHECK, ivykis >= $IVYKIS_UPDATED_VERSION, [IVYKIS_VERSION_UPDATED="yes"], [IVYKIS_VERSION_UPDATED="no"])
+    with_ivykis="system"
+    PKG_CHECK_MODULES(IVYKIS, ivykis >= $IVYKIS_MIN_VERSION,,)
+    PKG_CHECK_MODULES(IVYKIS_VERSION_CHECK, ivykis >= $IVYKIS_UPDATED_VERSION, [IVYKIS_VERSION_UPDATED="yes"], [IVYKIS_VERSION_UPDATED="no"])
 
-	# in case we're using a system installed ivykis, we can link against
-	# it even without libtool and without extra linker arguments (as
-	# we're linking dynamically)
+    # in case we're using a system installed ivykis, we can link against
+    # it even without libtool and without extra linker arguments (as
+    # we're linking dynamically)
 
-	IVYKIS_NO_LIBTOOL_LIBS="$IVYKIS_LIBS"
+    IVYKIS_NO_LIBTOOL_LIBS="$IVYKIS_LIBS"
 
-	dnl check if iv_work_pool_submit_continuation is available in the system ivykis
-	old_LIBS=$LIBS
-	LIBS="$LIBS $IVYKIS_LIBS"
-	AC_CHECK_FUNCS(iv_work_pool_submit_continuation, [], [AC_DEFINE(HAVE_IV_WORK_POOL_SUBMIT_CONTINUATION, 0)])
-	LIBS=$old_LIBS
+    dnl check if iv_work_pool_submit_continuation is available in the system ivykis
+    LIBS="$LIBS $IVYKIS_LIBS"
+    AC_CHECK_FUNC(iv_work_pool_submit_continuation,
+                    [AC_DEFINE([HAVE_IV_WORK_POOL_SUBMIT_CONTINUATION], [1], [Define if iv_work_pool_submit_continuation() exists])],
+                    [AC_DEFINE([HAVE_IV_WORK_POOL_SUBMIT_CONTINUATION], [0], [Define if iv_work_pool_submit_continuation() exists])])
+    LIBS=$LIBS_SAVE
```

Comment:
This isn't necessary. `SAFE_C_CHECK_END` will restore it two lines below.

Reply:
indeed, but if anytime any new code will be added in these if/else branches, it is much safer to restore the changes to eliminate future errors

as you can see, I used 

SAFE_C_CHECK_BEGIN/SAFE_C_CHECK_END

solely if the inner changes were simple (mainly when there is no if/else/swich constructs)